src/twisted/application/runner/_pidfile.py

```diff
@@ -0,0 +1,270 @@
+# -*- test-case-name: twisted.application.runner.test.test_pidfile -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+PID file.
+"""
+
+import errno
+from os import getpid, kill
+
+from zope.interface import Interface, implementer
+
+from twisted.logger import Logger
+
+
+
+class IPIDFile(Interface):
+    """
+    PID file.
+
+    Manages a file that remembers a process ID.
+    """
+
+    def read():
+        """
+        Read the process ID stored in this PID file.
+
+        @return: The contained process ID.
+        @rtype: L{int}
+
+        @raise NoPIDFound: If this PID file does not exist.
+        @raise EnvironmentError: If this PID file cannot be read.
+        @raise ValueError: If this PID file's content is invalid.
+        """
+
+
+    def write(pid):
```

Comment:
This interface seems a bit wide to me.  What's the use-case for writing some _other_ process's running PID?  Surely if you needed to do that for some reason you could construct a concrete `ForeignProcessPID` with the other process's PID, or parameterize the concrete `os.getpid` or somseuch.